server/static/dashboard/collab.js

```diff
@@ -7,42 +10,486 @@ collab.fetch(function(err) {
   document.querySelector('#partners').textContent = collab.data.users.slice().sort().join(' & ')
 });
 
+// Parameters required to use updateFunction
+// TODO: Reconcile parameters and extraArgs
+var parameters = {};
+
+// Whether deleted code is currently shown or not
+var showDeletedCode = false;
+
+
+//////////////////////////////////////
+///////////// MAIN CODE //////////////
+
+/**
+ * Fetch the files for this collab, determine which visual to use,
+ *   and display the files accordingly.
+ */
 connection.createFetchQuery('files', { collabid: collabid }, {}, function(err, files) {
   if (err) { throw err; }
-  
+
+  // Visual 1: total diff
```

Comment:
Instead of an enum of visualizations, make each option a boolean parameter: total diff on/off, regex on/off, etc.

Reply:
Done. Meant we don't need global state anymore.